OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs

```diff
@@ -300,14 +303,32 @@ void StartMissionClicked()
 				PlayVideo(fsPlayer, gameStartVideo, PlayingVideo.GameStart, () =>
 				{
 					StopVideo(fsPlayer);
-					Game.StartMission(selectedMapPreview.Uid, gameSpeed, difficulty, onStart);
+					StartMission();
 				});
 			}
 			else
-				Game.StartMission(selectedMapPreview.Uid, gameSpeed, difficulty, onStart);
-		}
+				StartMission();
+        }
+
+		void StartMission()
+		{
+			OrderManager om = null;
+
+			Action lobbyReady = null;
+			lobbyReady = () =>
```

Comment:
The reason its difficult is that the server sets up a large section of the lobby info for me including map,slots, and clients. So while its easy to make that work for restarting as there is already a lobby info around at first start there is not. My understanding of lambda functions is that whats inside of them is from local scope whenever they are created. This means I can't get the current lobby info inside of them or atleast last night I couldn't think of a nice way of doing it.